lib/core/resources/injector.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:dio/dio.dart';
+import 'package:get_it/get_it.dart';
```

Comment:
**question**: could you explain why you have decided to use use `get_it` instead of `BuildContext` for injecting data?

Reply:
I could have used `BuildContext` using `InheritedWidget`, to be honest, I did not do that because I know it's harder than just using `get_it`. Not that I don't want to do the harder things 😄 but I focused on other stuff with the time I had to develop the challenge.

But for sure we could use the context to inject dependencies using InheritedWidget and not depending on third-package libraries! With enough time it's possible to set up all the injecting system with the native stuff from Flutter.